contracts/extensions/bridge-operator-governance/BridgeManager.sol

```diff
@@ -0,0 +1,464 @@
+  }
+
+  /**
+   * @dev Internal function to add bridge operators.
+   *
+   * This function adds the specified `bridgeOperators` to the bridge operator set and establishes the associated mappings.
+   *
+   * Requirements:
+   * - The caller must have the necessary permission to add bridge operators.
+   * - The lengths of `voteWeights`, `governors`, and `bridgeOperators` arrays must be equal.
+   *
+   * @param voteWeights An array of uint256 values representing the vote weights for each bridge operator.
+   * @param governors An array of addresses representing the governors for each bridge operator.
+   * @return addeds An array of boolean values indicating whether each bridge operator was successfully added.
+   */
+  function _addBridgeOperators(
+    uint256[] memory voteWeights,
+    address[] memory governors,
+    address[] memory bridgeOperators
+  ) internal nonDuplicate(governors.extend(bridgeOperators)) returns (bool[] memory addeds) {
+    uint256 length = bridgeOperators.length;
+    if (!(length == voteWeights.length && length == governors.length)) revert ErrLengthMismatch(msg.sig);
+    addeds = new bool[](length);
+
+    EnumerableSet.AddressSet storage _governorSet = _getGovernorsSet();
+    mapping(address => address) storage _governorOf = _getGovernorOf();
+    EnumerableSet.AddressSet storage bridgeOperatorSet = _getBridgeOperatorSet();
+    mapping(address => BridgeOperatorInfo) storage _governorToBridgeOperatorInfo = _getGovernorToBridgeOperatorInfo();
+
+    // get rid of stack too deep
+    uint256 accumulatedWeight;
+    {
+      address governor;
+      address bridgeOperator;
+      BridgeOperatorInfo memory bridgeOperatorInfo;
+
+      for (uint256 i; i < length; ) {
+        governor = governors[i];
+        bridgeOperator = bridgeOperators[i];
+
+        _requireNonZeroAddress(governor);
+        _requireNonZeroAddress(bridgeOperator);
+        _requirePayableAddress(bridgeOperator);
+
+        addeds[i] = bridgeOperatorSet.add(bridgeOperator);
+
+        if (addeds[i]) {
+          _governorSet.add(governor);
+
+          if (voteWeights[i].toUint96() == 0) revert ErrInvalidVoteWeight(msg.sig);
+
+          // get rid of stack too deep
+          // bridgeOperatorInfo.voteWeight = voteWeights[i].toUint96();
+          // accumulatedWeight += bridgeOperatorInfo.voteWeight
+          accumulatedWeight += bridgeOperatorInfo.voteWeight = voteWeights[i].toUint96();
+          _governorOf[bridgeOperator] = governor;
+          bridgeOperatorInfo.addr = bridgeOperator;
+          _governorToBridgeOperatorInfo[governor] = bridgeOperatorInfo;
+        }
+
+        unchecked {
+          ++i;
+        }
+      }
+    }
+
+    TOTAL_WEIGHTS_SLOT.addAssign(accumulatedWeight);
+
+    _notifyRegisters(IBridgeManagerCallback.onBridgeOperatorsAdded.selector, abi.encode(bridgeOperators, addeds));
+
+    emit BridgeOperatorsAdded(addeds, voteWeights, governors, bridgeOperators);
```

Comment:
Does it mean when the value in `added` is false, the other fields with the same index don't take effect?